File: packages/ui-flutter/lib/src/context_menu.dart
import 'package:flutter/material.dart';

import 'theme.dart';

/// Mirrors `packages/ui/src/components/context-menu.tsx`. Wraps a
/// `MenuAnchor` opened by long-press (mobile) or secondary-tap (desktop).
/// Put `KinetixMenuItem` / `KinetixMenuSeparator` / `KinetixMenuLabel`
/// (from dropdown_menu.dart) in `menuChildren`.
class KinetixContextMenu extends StatelessWidget {
  const KinetixContextMenu({super.key, required this.menuChildren, required this.child});

  final List<Widget> menuChildren;
  final Widget child;

  @override
  Widget build(BuildContext context) {
    final c = KinetixTheme.of(context);
    return MenuAnchor(
      style: MenuStyle(
        backgroundColor: WidgetStatePropertyAll(c.popover),
        surfaceTintColor: const WidgetStatePropertyAll(Color(0x00000000)),
        shape: WidgetStatePropertyAll(
          RoundedRectangleBorder(
            borderRadius: BorderRadius.circular(8),
            side: BorderSide(color: c.border),
          ),
        ),
      ),
      menuChildren: menuChildren,
      builder: (context, controller, _) => GestureDetector(
        behavior: HitTestBehavior.opaque,
        onLongPress: controller.open,
        onSecondaryTapDown: (_) => controller.open(),
        child: child,
      ),
    );
  }
}

File: packages/ui-swiftui/Sources/KinetixUI/ContextMenu.swift
//
// ContextMenu.swift — KinetixContextMenu.
//
// Mirrors packages/ui/src/components/context-menu.tsx. Wraps SwiftUI's
// native `.contextMenu` (long-press on iOS, right-click on macOS). Put
// `KinetixMenuItem` / `KinetixMenuSeparator` / `KinetixMenuLabel` (from
// DropdownMenu.swift) in the `menu` builder — same shared row parts.
//

import SwiftUI

public struct KinetixContextMenu<Content: View, MenuItems: View>: View {
    private let content: Content
    private let menuItems: MenuItems

    public init(
        @ViewBuilder content: () -> Content,
        @ViewBuilder menu: () -> MenuItems
    ) {
        self.content = content()
        self.menuItems = menu()
    }

    public var body: some View {
        content.contextMenu {
            menuItems
        }
    }
}
